p2p/transport_test.go

```diff
@@ -638,6 +641,25 @@ func TestTransportAddChannel(t *testing.T) {
 	}
 }
 
+func TestTransportResolveIPs(t *testing.T) {
+	server := startFakeDNS()
+
+	r := &net.Resolver{
+		PreferGo: true,
+		Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+			d := net.Dialer{
+				Timeout: time.Millisecond * time.Duration(500),
+			}
+			return d.DialContext(ctx, network, "127.0.0.1:5355")
+		},
+	}
+	_, err := resolveIPs(r, &testTransportConn{})
+	require.Contains(t, err.Error(), "lookup test.local: i/o timeout")
+
+	err = server.Shutdown()
+	require.NoError(t, err)
```

Comment:
Could you please apply a style that immediately after allocating a resource, releases the resource with `defer`? Specifically, I'd like these two-lines to be `defer` and move to the line after `server := startFakeDNS()`.